### PR TITLE
Various Improvements from GTNH Fork

### PR DESCRIPTION
**Pull request by glowredman**
This PR intends to make the GTNH fork redundant, allowing the GTNH pack to use this fork and development efforts to be concentrated here.

See commit history for other changes (not necessarily based on the GTNH fork).

### CODE REVIEW

**Comment by jss2a98aj** [comment]
The squid ink cloud backport would fit better in Et Futurum Requium, which is currently being considered for inclusion in GTNH. The rest of the changes are definately fine though.

**Comment by mitchej123** [comment]
+1 to moving this to EFR and pulling EFR into GTNH

Thanks for helping sync the forks up!

**Comment by Dream-Master** [comment]
if this is ported we can use upstream of Bug torch ?

**Comment by glowredman** [comment]
> The squid ink cloud backport would fit better in Et Futurum Requiem

I see what I can do.
> if this is ported we can use upstream of Bug torch ?

Yes, that's the main intention.

**Comment by glowredman** [comment]
@jss2a98aj Done :)

File: src/main/java/jss/bugtorch/mixinplugin/BugTorchEarlyMixins.java [hunk 1]
```diff
@@ -36,6 +38,8 @@ public List<String> getMixins(Set<String> loadedCoreMods) {
             BugTorch.logger.info("NotFine detected, skipping redundant early mixins.");
             useNotFineOverlap = false;
         }
+        
+        txLoaderPresent = client && loadedCoreMods.contains("glowredman.txloader.TXLoaderCore");
```
Comment (jss2a98aj) [inline src/main/java/jss/bugtorch/mixinplugin/BugTorchEarlyMixins.java:42]: The ItemFireball mixin should probably not apply if this is false, right?

**Review by jss2a98aj — COMMENTED** [review]
Sorry for taking so long to get to this. Most of the changes look good.

**Review by jss2a98aj — APPROVED** [review]

**Pull request merged**